src/main/java/de/thomaskrille/dropwizard_template_config/TemplateConfigurationSourceProvider.java

```diff
@@ -69,8 +75,15 @@ public InputStream open(final String path) throws IOException {
 
             new Template("config", configTemplate, configuration)
                     .process(dataModel, new OutputStreamWriter(processedTemplateStream, charset));
+            byte[] processedTemplateBytes = processedTemplateStream.toByteArray();
 
-            return new ByteArrayInputStream(processedTemplateStream.toByteArray());
+            if (outputPath.isPresent()) {
+                File outputFile = new File(outputPath.get());
+                Files.createParentDirs(outputFile);
+                Files.write(processedTemplateBytes, outputFile);
```

Comment:
It's not entirely clear to me how Dropwizard will handle an error here. Catching the error and logging sounds like a reasonable option, but I don't think logging in an option here since this code is getting executed before the configuration is processed, so logging isn't set up yet and the messages won't be going anywhere.